### ISSUE TITLE
Option to use underlined headings

### ISSUE THREAD

**Issue by casey** [kivikakk/comrak#137]
I prefer to use the underlined form for H1 and H2 headings, like so:

```markdown
h1 heading
==========

h2 heading
----------
```

I personally find that it sets them off a bit more, and makes it easier to see the structure of a document at a glance, when reading the raw markdown.

What would you think of an option to use underlined H1 and H2 headings when formatting a tree as CM?

This seems like a simple change, so if you're amenable to it, I'd like to give implementing it a shot.

**Comment by kivikakk** [comment]
Absolutely! Per #136, consider refactoring a bit so we can group output formatting options.

**Comment by casey** [comment]
I hacked together a super-gross regex-based solution for my original use case, so until it falls apart, I probably won't be able to get to this.
